File: Java/dependency-injection/bastard-injection/src/main/java/com/github/grzesiek_galezowski/ChangingBehaviorThroughComposition/SwitchCriterias/MetOnWorkDayCriteria.cs
using System;

namespace ChangingBehaviorThroughComposition
{
  public class MetOnWorkDayCriteria : SwitchCriteria
  {
    public bool IsNotMet()
    {
      return IsWeekend();
    }

    private bool IsWeekend()
    {
      var dayOfWeek = DateTime.Now.DayOfWeek;
      return dayOfWeek == DayOfWeek.Saturday || dayOfWeek == DayOfWeek.Sunday;
    }
  }
}

File: Java/dependency-injection/changing-behavior-through-composition/src/main/java/com/github/grzesiek_galezowski/ChangingBehaviorThroughComposition/SwitchCriterias/MetOnWorkDayCriteria.java
package com.github.grzesiek_galezowski.ChangingBehaviorThroughComposition.SwitchCriterias;

import com.github.grzesiek_galezowski.ChangingBehaviorThroughComposition.SwitchCriteria;

import java.time.DayOfWeek;
import java.time.LocalDate;

public class MetOnWorkDayCriteria implements SwitchCriteria {
  public boolean isNotMet() {
    return isWeekend();
  }

  private boolean isWeekend() {
    DayOfWeek dayOfWeek = LocalDate.now().getDayOfWeek();
    return dayOfWeek.equals(DayOfWeek.SATURDAY) || dayOfWeek.equals(DayOfWeek.SUNDAY);
  }
}
